src/main/webapp/app/shared/virtual-scroll/virtual-scroll.component.ts

```diff
@@ -0,0 +1,255 @@
+/**
+ * Component based on Open Source Project ag-virtual-scroll
+ * https://github.com/ericferreira1992/ag-virtual-scroll
+ *
+ */
+
+import { Component, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, SimpleChanges, ViewChild } from '@angular/core';
+import { NavigationStart, Router } from '@angular/router';
+import { VirtualScrollRenderEvent } from 'app/shared/virtual-scroll/virtual-scroll-render-event.class';
+import { ITEMS_PER_PAGE } from 'app/shared/constants/pagination.constants';
+
+@Component({
+    selector: 'virtual-scroll',
+    templateUrl: './virtual-scroll.component.html',
+    styleUrls: ['virtual-scroll.style.css'],
+})
+export class VirtualScrollComponent implements OnInit, OnChanges, OnDestroy {
+    @ViewChild('itemsContainer', { static: true }) private itemsContainerElRef: ElementRef<HTMLElement>;
+
+    @Input('items') public originalItems: any[] | undefined = [];
```

Comment:
can we use generics here as well?

Reply:
I do not think the use of them will be advantageous since the component had to be highly customized to work for the communication overview page.

Reply:
I suggest that we use Angular's way for virtual scrolling in other places if the items are of fixed height.

Reply:
Generics are added ✔️